### PR TITLE
Allow user to add github team

### LINKED ISSUE #226: Use GitHub Teams to manage seal members instead of hard-coding
This is a feature request.

Currently updating the GitHub users who should be included in a team's seal posts is a manual process, and involves duplication; teams have to ensure that their members are updated in both GitHub teams and in the seal. In most circumstances the list of members should be the same for both, and it should be possible for the seal to [query a public team's members using the GitHub API and the team name](https://docs.github.com/en/rest/reference/teams#list-team-members).

### PR DESCRIPTION
**Pull request by DavidBiddle**
Closes #226 
This allows the user to add a github team in addition to or instead of the existing members and include_repos settings.

I'm planning to follow this up later with a PR adding the forms team to the config, which should make it a bit clearer how this works, but also hoping my documentation changes here are clear enough.

### CODE REVIEW

File: lib/github_fetcher.rb
Comment (ChrisBAshton) [inline lib/github_fetcher.rb:134]: I'd move the presence check to this method, so that it uses the `team_slug` parameter rather than just passing it to `get_github_team_members`. Also not sure under what circumstances `team_slug` would evaluate to an empty string rather than `nil`. Best to leave that bit out unless we know we need it.

```suggestion
    return members if team_slug.nil?

    github_team_members = get_github_team_members(team_slug)
```

File: lib/github_fetcher.rb
Comment (ChrisBAshton) [inline lib/github_fetcher.rb]:
```suggestion
    github.get("/orgs/#{@organisation}/teams/#{team_slug}/members").map { |member| member.login }
```

File: lib/github_fetcher.rb
Comment (ChrisBAshton) [inline lib/github_fetcher.rb:146]: Same comment here, but for repos rather than members.

**Review by ChrisBAshton — COMMENTED** [review]
Nice work on this! ⭐ The changes look great - just a couple of minor comments below. There's also now a file conflict that needs addressing.

Otherwise, it looks good to go! I do wonder what the knock-on might be for GitHub rate limits; this change means we're making more API requests, so it could be that we need to build in some `sleep`s to slow down the requests. Difficult to know until it's merged and being used though.

**Comment by DavidBiddle** [comment]
Thanks @ChrisBAshton - good suggestions! I've made those changes now 🙂

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍 thanks for making the changes.

**Comment by DavidBiddle** [comment]
Thanks again @ChrisBAshton - I'm afraid I can't merge as I don't have write permissions, please could you do the honours?

**Pull request merged**
